### PR TITLE
Starfix cli blob patch

### PR DESCRIPTION
**Pull request by fahad-israr**
- Support added for opening file to specific line number.
- Handled all three IDEs separately.
- Tested to work on VS Code (on Linux  ) .
- Fixes  [#61](https://github.com/starfixdev/starfix/issues/61)

### CODE REVIEW

**Review by maxandersen — CHANGES_REQUESTED** [review]

**Comment by fahad-israr** [comment]
- Doesn't work on Eclipse and Idea .
- Splitting on basis of spaces as separate command doesn't works well if you have file name with spaces in it. Have a look at URL like https://github.com/maxandersen/reallyevil/blob/main/here%20is%20a%20space.txt . Avoid manual parsing : instead use the URL Object for parsing the URL.
- Instead of embedding everything here itself try creating a new ide class/interface that does the processing as needed in specific cases .

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 1]
```diff
@@ -86,11 +87,33 @@ public int cloneCmd(@Parameters(index = "0") String url) {
...
 
                 String temp = url.substring(url.indexOf("blob/")+5);
                 branch = temp.substring(0,temp.indexOf("/"));
                 filePath = temp.substring(temp.indexOf("/")+1);
                 url = url.substring(0,url.indexOf("/blob"));
+
+                if(filePath.indexOf("#")>0){
+                    // ==== VS CODE ====
+                    // code -g file:line
+                    if(ide.equals("code")||ide.equals("code.cmd")){
```
[the remark this answers is not in the document]
Comment (fahad-israr) [reply cli/src/main/java/dev/starfix/Starfix.java]: Moved to a separate class currently and would gradually improve the structure after successfully implementing the intended functionality. Currently focussing on getting the editors to open up and point to expected line numbers.

**Comment by fahad-israr** [comment]
- Now Works on IntelliJ Idea too. Tested on Linux(Ubuntu 20.x)

**Comment by fahad-israr** [comment]
@maxandersen  I was successful in getting the line number thing to work on IntelliJ and Vs Code. I have been trying for eclipse but not much luck. 
I am tring to run `eclipse project_path file_path:line_number` but it isn't working . I am not sure if its not working only on my system specifically due to some issue or it doesn't work at all. 
I may a little hint from you here.

**Comment by maxandersen** [comment]
It might be is specific. See https://bugs.eclipse.org/bugs/show_bug.cgi?id=305336

' --launcher.openFile c:\workspace\MyClass.java:50' 

Seem to be whats needed on Linux if calling eclipse directly.

File: cli/src/main/java/dev/starfix/Editor.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package dev.starfix;
+
+import java.nio.file.Path;
+import java.io.IOException;
+import static dev.starfix.Starfix.runCommand;
+
+
+public class Editor{
+
+    public static void launch_editor(Path directory, String ide, String path, String filePath) throws IOException, InterruptedException {
```
Comment (maxandersen) [inline cli/src/main/java/dev/starfix/Editor.java:10]: moving static method into a new class doen't change much - just adds another file but keep same coupling. 

should be a non-static method and have class per unique enough cases of IDE's

**Comment by maxandersen** [comment]
mrging this as code works for more cases than current

**Pull request merged**